highlight/highlight
# Support python requests autoinstrumentation

## Summary
Add tracing auto-instrumentation for the requests python library

## How did you test this change?
1) Start the e2e flask app
```
cd sdk/highlight-py/e2e/highlight_flask
poetry build
poetry run flask run
```
2) Hit the app via curl
```
curl -H "x-highlight-request: bs1/br1" http://localhost:5000
```
3) Confirm that the `GET` trace for the requests library is created and linked to the same trace as the parent span

![Screenshot 2023-12-12 at 1 46 17 PM](https://github.com/highlight/highlight/assets/17744174/832b8ed5-9863-4f0e-b2ef-4a133c0b2ee0)

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
N/A
